### DISCUSSION TITLE
Integration request with https://sourcify.dev/

### DISCUSSION THREAD

**Discussion by hackfisher** [ringdao-community-guild/discussion#116] [Integration]
Adding Darwinia/Crab chain and Pangolin Testnet

**Comment by hackfisher** [comment]
https://docs.sourcify.dev/docs/chain-support/

**Reply by nblogist** [reply]
>If you are not able to deploy new test contracts above (e.g. because it costs on a mainnet) you can provide existing contracts. Create a folder services/server/test/chains/sources/{chainId} and store the contract source files and metadata of already deployed contracts there.

@hackfisher can you please provide the contract source file and metadata file.

**Reply by nblogist** [reply]
@wuminzhe

**Reply by hamidroohi71** [reply]
Hey. As I know and see in the docs we should just use a contract already deployed into the chain and put its metadata file beside it. Also, we should set this contract address in the chain process. 
I did it but I got the error: Response body: {"error":"The deployed and recompiled bytecode don't match."}. 
Any idea about it?
Mention this point that I generated the metadata files by the use of remix IDE and I think the data in it is different in the metadata received at the deployed time

**Reply by wuminzhe** [reply]
@AurevoirXavier Can you help this?

**Comment by hackfisher** [comment]
The issue has been recommended for escalation and will be followed up by Echo. @hujw77

**Reply by hujw77** [reply]
https://github.com/ethereum/sourcify/pull/1444
